mne/decoding/base.py

```diff
@@ -723,3 +723,72 @@ def _check_estimator(estimator, get_params=True):
         raise ValueError('estimator must be a scikit-learn transformer or an '
                          'estimator with the get_params method that allows '
                          'cloning.')
+
+def _check_estimator(estimator, get_params=True):
+    """Check whether an object has the fit, transform, fit_transform and
+    get_params methods required by scikit-learn"""
+    valid_methods = ('predict', 'transform', 'predict_proba',
+                     'decision_function')
+    if (
+        (not hasattr(estimator, 'fit')) or
+        (not any(hasattr(estimator, method) for method in valid_methods))
+    ):
+        raise ValueError('estimator must be a scikit-learn transformer or '
+                         'an estimator with the fit and a predict-like (e.g. '
+                         'predict_proba) or a transform method.')
+
+    if get_params and not hasattr(estimator, 'get_params'):
+        raise ValueError('estimator must be a scikit-learn transformer or an '
+                         'estimator with the get_params method that allows '
+                         'cloning.')
+
+
+def _get_final_est(estimator):
+    """Return the final component of a sklearn estimator/pipeline.
+
+    Parameters
+    ----------
+    estimator : pipeline, or sklearn / MNE estimator
+        An estimator chain from which you wish to pull the final estimator.
+
+    Returns
+    -------
+    estimator : the sklearn-style estimator at the end of the input chain.
+    """
+    # Define classes where we pull `estimator` manually
+    import sklearn
+    from distutils.version import LooseVersion
+    if LooseVersion(sklearn.__version__) < '0.16':
+        raise ValueError('sklearn support requires sklearn version >= 0.16')
+
+    attributes = [ii for ii in ['_final_estimator', 'base_estimator']
+                  if hasattr(estimator, ii)]
+    if len(attributes) > 0:
+        # In the event that both are present, `_final_estimator` is prioritized
+        estimator = getattr(estimator, attributes[0])
+        estimator = _get_final_est(estimator)
+    return estimator
+
+
+def get_coefs(estimator, coef_name='coef_'):
```

Comment:
Yeah I think you make a good point. I guess the question is: "is it better to leave this code as-is and have flexibility in the estimators used in the future, or better to take this code out and have a soft requirement that future estimators must support a `coef_` attribute"? I can go either way on it because I agree w/ you about not including unnecessary code.